app/controllers/users_controller.rb

```diff
@@ -0,0 +1,39 @@
+class UsersController < ApplicationController
+  before_action :require_login, only: [:dashboard]
+  before_action :registered_user, only: [:new]
+
+  def index
+    @users = User.all
+  end
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(create_params[:user])
+
+    if @user.save
+        redirect_to login_url
+    else
+      redirect_to signup_path
+    end
+  end
+
+  def show
+    @user = User.find(params[:id])
+  end
+
+  def dashboard
+    @user = User.find(session[:user_id])
+    @products = @user.products
+    @orders = Order.includes(:order_items).where(order_items: { user_id: @user } )
```

Comment:
I attempted to pull the @orders logic out into a model scope or Orders, however then I'm not sure how to access @user anymore. Ideally, we would have set up associations so that we could just call User.orders to find all relevant orders, that would be a much larger refactoring of our schema.